File: raeunlee/Week1/01.04/PGS_입국심사.py
def solution(n, times):
    answer = 0
    times.sort()
    start = times[0]
    end = times[-1] * n
    while start <= end:
        people = 0 # 입국 심사 완료된 사람 수
        mid = (start + end) // 2 # 중앙값
        for time in times:
            # 중앙값 동안 입국 심사가 가능한 사람 수
            people += mid//time
        # n 이상 심사할 경우 일단 answer 에 저장하고 최소 mid를 찾기
        if people >= n:
            answer = mid
            end = mid - 1
        #n보다 적게 심사, mid로는 부족
        else:
            start = mid + 1
    return answer

File: hjkim/Week1/01.04/PGS_입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        Arrays.sort(times);

        long start = times[0];
        long end = (long) n * times[times.length - 1];
        long answer = end;

        while (start <= end) {
            long mid = (start + end) / 2;
            long cnt = 0; // 심사 가능한 사람 수
            for (int t : times) {
                cnt += mid / t; // 각 심사관이 심사할 수 있는 사람 수
                if(cnt >= n) break;
            }

            if (cnt < n) {
                start = mid + 1;
            } else {
                end = mid - 1;
                answer = mid;
            }
        }
        return answer;
    }
}